CRM/Activity/BAO/Activity.php

```diff
@@ -2384,7 +2384,7 @@ public static function getContactActivitySelector(&$params) {
     $viewOnlyCaseActivityTypeIDs = array_flip(CRM_Activity_BAO_Activity::getViewOnlyActivityTypeIDs());
 
     if (!empty($activities)) {
-      $activityStatus = CRM_Core_PseudoConstant::activityStatus();
+      $activityStatus = CRM_Activity_BAO_Activity::buildOptions('status_id', 'get');
```

Comment:
These aren't quite the same. The original excludes disabled statuses, the replacement includes them. Ideally, we'd review each one carefully to see if the original made sense or whether it should include disabled, but then that will stall. (In this particular spot, the before gives an E_NOTICE and to me looks wrong because in the selector it shows as no status, so I like the replacement. But others will disagree.)

Reply:
Ah yes. Looking deeper into the code, I think we can use 'search' instead of 'get' for status. The only change seems to be it excludes inactive options and inactive components' options. At least for activity status and type.

Reply:
I've changed this to 'search' and in a few other places.